### ISSUE TITLE
[java] BigIntegerInstantiation false negative with compile-time constant

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4158]
Besides, I found the rule  [BigIntegerInstantiation](https://pmd.github.io/latest/pmd_rules_java_performance.html#bigintegerinstantiation) also has similar problem. Please read the code example below:
```java
import java.math.BigInteger;
public class Foo {
  final String str = "1";
  BigInteger b = new BigInteger(str);  // should report a warning at this line, but not
}
```
PMD should have reported a warning at line 4, because there is a BigInteger defintion same with predefined instances. So, this is a false negative.

_Originally posted by @Belle-PL in https://github.com/pmd/pmd/issues/3847#issuecomment-1235151278_
